index.js

```diff
@@ -1,28 +1,84 @@
 
-setInterval(function() {
-  if (typeof window.web3 !== "undefined") {
-    if (window.web3.eth.accounts[0]) {
-      window.location.reload();
+const FORWARDER_TYPE = {
+  IFRAME: 'IFRAME',
+  TAB: 'TAB',
+}
+
+class Forwarder {
+  constructor () {
+    this.refreshInterval = 3000
+    this.parentMessageInterval = 1000
+
+    this.initialize = this.initialize.bind(this)
+    this._receiveMessage = this._receiveMessage.bind(this)
+    this._receiveParentMessage = this._receiveParentMessage.bind(this)
+  }
+
+  initialize () {
+    if (!Forwarder.isMetaMaskInstalled()) {
+      return setTimeout(() => window.location.reload(), this.refreshInterval)
+    }
+
+    if (window.opener) {
+      this.parentWindow = window.opener
+      this.forwarderType = FORWARDER_TYPE.TAB
+    } else if (window.parent) {
+      this.parentWindow = window.parent
```

Comment:
So I assume the `this.parentWindow = window.parent` reference isn't broken by the parent window reloading?

Reply:
Good question - I don't believe so. Worth verifying maybe, but it's only relevant for the case where a tab is used instead of an iframe, and that flow is not great anyway (requires allowing a pop-up).

Reply:
I can't speak to whether or not this reference is "broken" after the parent reloads, but I can say that the condition here, `else if (window.parent)`, will almost always be truthy as `window.parent === window` for new windows.

Reply:
Ah right, good observation. I'll update this.

Reply:
The condition has been updated to compare `window.parent` to `window` to determine whether it's in an iframe. Preliminary testing seems to indicate that this works.

I also tested whether `window.parent` was correct after reloading the iframe. It is - the reference is preserved through reloading.